ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -58,56 +59,108 @@ func NewReadWriter() (*YAMLReadWriter, error) {
 	return &YAMLReadWriter{destination: dest}, nil
 }
 
-func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error) {
+func readINI(dest *Destination) (*CLIConfig, error) {
 	iniReadWriter, err := NewINIReadWriter(dest)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	return iniReadWriter.GetConfig()
 }
 
-func readYAML(yamlPath string, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
-	// convert yaml to CliConfig
-	dat, err := ioutil.ReadFile(yamlPath)
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readClusterConfig(path string, clusterConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read config file: "+path)
+	}
+
+	config := ClusterConfig{Clusters: make(map[string]Cluster)}
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
+	}
+
+	// get the correct cluster
+	chosenCluster := ""
+	if clusterConfigKey == "" {
+		chosenCluster = config.Default
+	} else {
+		chosenCluster = clusterConfigKey
+	}
+	cluster := config.Clusters[chosenCluster]
+
+	// Get the info out of the cluster
+	cliConfig.Region = cluster.Region
+	cliConfig.Cluster = cluster.Cluster
+	cliConfig.ComposeServiceNamePrefix = cluster.ComposeServiceNamePrefix
+	return nil
+
+}
+
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readProfileConfig(path string, profileConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
```

Comment:
Incorrect comment.

Looks like there are a lot of shared contents between `readProfileConfig` (line 103-118) and `readClusterConfilg` (line 73-89). Can we create a helper function for this?